public/app/features/canvas/runtime/ables.tsx

```diff
@@ -38,6 +36,16 @@ export const settingsViewable = (scene: Scene) => ({
           const evt = new PointerEvent('contextmenu', { clientX: event.clientX, clientY: event.clientY });
           container.dispatchEvent(evt);
         }}
+        onKeyDown={(event) => {
+          if (event.key === 'Enter') {
+            const container = moveable.getContainer();
+            const rect = event.currentTarget.getBoundingClientRect();
+            const evt = new PointerEvent('contextmenu', { clientX: rect.x, clientY: rect.y });
+            container.dispatchEvent(evt);
```

Comment:
These changes look great! It might make sense though to avoid duplicating the onClick logic and breaking it out into a separate `onSettingsClick` function that we can reference for both the `onClick` and `onKeyDown` props 👍

Reply:
I also thought about it but they have a bit different body function and I decided to keep it separated. Do you have any idea how to optimize it?

Reply:
Yeah, I think we can put it in a separate function, just to do different calculations depending on the event type.

Reply:
I was thinking of just consolidating the shared logic more along the lines of
```
const openSettings = () => {
    const container = moveable.getContainer();
    const evt = new PointerEvent('contextmenu', { clientX: event.clientX, clientY: event.clientY });
    container.dispatchEvent(evt);
}
```
where
`onClick={openSettings}` 
and
`onKeyDown={() => { key check {openSettings()}}`

Reply:
ah just noticed the difference - yeah should be able to just pass even in and do that logic in the separate function 👍

Reply:
There is a trick with this function, I can't just set two types for event e.g. `event: PointerEvent | KeyboardEvent` or `SyntheticEvent`.

```
const onSettingsClick = (event: any) => {
      let clientX = 0;
      let clientY = 0;

      if (event.type === 'click') {
        clientX = event.clientX;
        clientY = event.clientY;
      } else if (event.type === 'keydown' && event.key === 'Enter') {
        const rect = event.currentTarget.getBoundingClientRect();
        clientX = rect.x;
        clientY = rect.y;
      }

      const container = moveable.getContainer();
      const evt = new PointerEvent('contextmenu', { clientX, clientY });
      container.dispatchEvent(evt);
    };
```

any suggestions here?

Reply:
hm what if we just broke this out with 3 functions:

```
const onClick = (event) => {
  openSettings(event.clientX, event.clientY); 
}

const onKeyPress = (event) => {
  const rect = event.currentTarget.getBoundingClientRect();
  openSettings(rect.x, rect.y);
}

const openSettings = (clientX: number, clientY: number) => {
      const container = moveable.getContainer();
      const evt = new PointerEvent('contextmenu', { clientX, clientY });
      container.dispatchEvent(evt);
    };
```

Reply:
Oh, cool, even better! Thanks!